packages/ag-charts-community/src/chart/cartesianChart.ts

```diff
@@ -314,9 +319,109 @@ export class CartesianChart extends Chart {
             });
         }
 
+        this.applyAxisCrossing(seriesRect, crossPositions);
+
         return { clipSeries, seriesRect, axisAreaWidths: newAxisAreaWidths, overflows };
     }
 
+    private calculateAxesCrossPositions(axisWidths: Map<string, number>, seriesRect: BBox): Map<string, number> {
```

Comment:
It might be better to short-circuit early if no axes have a `crossAt?.value` - e.g. I would make the first step here to find all axes configured in that way, before we get into searching for X and Y axes.

If you could do this check earlier outside of iterating on the axis layout, it would be even better to minimise redundant checks on each iteration.